addons/web/static/src/js/core/sorted_registry.js

```diff
@@ -0,0 +1,64 @@
+odoo.define('web.SortedRegistry', function (require) {
+"use strict";
+
+var Registry = require('web.Registry');
+
+/**
+ * A sorted registry is a registry in whick items are sorted.
+ */
+var SortedRegistry = Registry.extend({
+    /**
+     * @constructor
+     * @param {Object} [mapping] the initial data in the registry
+     */
+    init: function () {
+        var self = this;
+        this._super.apply(this, arguments);
+        this._scores = [];
+        this._initialKeys = Object.keys(this.map);
+        this._sortedKeys = this._initialKeys;
+        this.listeners.push(function () {
```

Comment:
this is complicated. why not override the keys method?  Also, why do you keep the initial keys unsorted? and why do you use the sortedKeys variable here when you define the _sortedKeys above?